contracts/Strategy.sol

```diff
@@ -10,38 +8,78 @@ import {
+import {IUniswapV2Router02} from "../interfaces/solidly/IUniswapV2Router02.sol";
 
 contract Strategy is BaseStrategy {
     using SafeERC20 for IERC20;
     using Address for address;
     using SafeMath for uint256;
 
-    constructor(address _vault) public BaseStrategy(_vault) {
-        // You can set these parameters on deployment to whatever you want
-        // maxReportDelay = 6300;
-        // profitFactor = 100;
-        // debtThreshold = 0;
+    // tokens
+    IERC20 internal constant USDT =
+        IERC20(0x049d68029688eAbF473097a2fC38ef61633A3C7A);
+    IERC20 internal constant nUSD =
+        IERC20(0xED2a7edd7413021d440b09D654f3b87712abAB66);
+    IERC20 internal constant SYN =
+        IERC20(0xE55e19Fb4F2D85af758950957714292DAC1e25B2);
+
+    // lp tokens
+    IERC20 internal immutable syn3PoolLP;
+
+    // staking contracts
+    IMasterChef internal immutable synStakingMC;
+
+    // pools
+    ISwap internal immutable syn3PoolSwap; // Synapse Fantom Stable 3 Pool
+
+    // router
+    IUniswapV2Router02 internal immutable solidlyRouter;
+
+    uint256 internal immutable pid; // Staking contract Pool ID
+    uint8 internal immutable syn3PoolUSDCTokenIndex; // Index of USDT in Synapse Fantom 3 Pool
+
+    constructor(
+        address _vault,
+        address _synStable3PoolLP,
+        address _synStable3Pool,
+        address _solidlyRouter,
+        address _synStakingMC,
+        uint256 _stakingContractPoolId,
+        uint8 _synStable3PoolUSDCTokenIndex
+    ) public BaseStrategy(_vault) {
+        minReportDelay = 60 * 60 * 24 * 7; // 7 days
+
+        syn3PoolLP = IERC20(_synStable3PoolLP); // FTM Mainnet: 0x2DC777ff99058a12844A33D9B1AE6c8AB4701F66
+        syn3PoolSwap = ISwap(_synStable3Pool); // FTM Mainnet: 0x85662fd123280827e11C59973Ac9fcBE838dC3B4
+        synStakingMC = IMasterChef(_synStakingMC); // FTM Mainnet: 0xaeD5b25BE1c3163c907a471082640450F928DDFE
+        solidlyRouter = IUniswapV2Router02(_solidlyRouter); // FTM Mainnet: 0xa38cd27185a464914D3046f0AB9d43356B34829D
+
+        pid = _stakingContractPoolId; // FTM Mainnet: 3
+        syn3PoolUSDCTokenIndex = _synStable3PoolUSDCTokenIndex; // FTM Mainnet: 1
     }
 
     // ******** OVERRIDE THESE METHODS FROM BASE CONTRACT ************
 
     function name() external view override returns (string memory) {
         // Add your own name here, suggestion e.g. "StrategyCreamYFI"
-        return "Strategy<ProtocolName><TokenType>";
+        return "StrategySynapseUSDC";
     }
 
     function estimatedTotalAssets() public view override returns (uint256) {
-        // TODO: Build a more accurate estimate using the value of all positions in terms of `want`
-        return want.balanceOf(address(this));
+        // Incase we aren't able to withdraw our deposit all in `want`, we might have other types of stable coin tokens
+        return
+            nusdBalance().add(usdtBalance()).add(wantBalance()).add(
```

Comment:
i would remove all this and make estimatedTotalAssets the sum of
- want value of the staked LPs
- want value of unstaked LPs
- add want.balanceOf(strategy)

we typically dont use the farmed token value in estimatedTotalAssets.